src/include/network/postgres_protocol_utils.h

```diff
@@ -265,14 +313,72 @@ class PostgresPacketWriter {
     BeginPacket(NetworkMessageType::SIMPLE_QUERY_COMMAND).AppendString(query).EndPacket();
   }
 
+  /**
+   * Writes an empty query response
+   */
+  void WriteEmptyQueryResponse() { BeginPacket(NetworkMessageType::EMPTY_QUERY_RESPONSE).EndPacket(); }
+
+  /**
+   * Writes a no-data response
+   */
+  void WriteNoData() { BeginPacket(NetworkMessageType::NO_DATA_RESPONSE).EndPacket(); }
+
+  void WriteParameterDescription(const std::vector<PostgresValueType> &param_types) {
+    BeginPacket(NetworkMessageType::PARAMETER_DESCRIPTION);
+    AppendValue<int16_t>(static_cast<int16_t>(param_types.size()));
+
+    for (auto &type : param_types) AppendValue<int32_t>(static_cast<int32_t>(type));
+
+    EndPacket();
+  }
+
+  /**
+   * Writes row description, as the first packet of sending query results
+   * @param columns the column names
+   */
+  void WriteRowDescription(const std::vector<std::string> &columns) {
```

Comment:
Is this information going to be provided by `ResultSet`? We should have the ExecEngine generate the exact object that this methods needs.

Reply:
Yes, because actually this command also needs a lot of other information like table OIDs. They should go to `ResultSet`. Perhaps this function should accept a `ResultSet`? 

The current implementation does not make a `Resultset` for describe command, so the API is like this...